server/server.js

```diff
@@ -153,141 +128,168 @@ app.post('/genotet/log', function(req, res) {
 app.post('/genotet/upload', upload.single('file'), function(req, res) {
   log.serverLog('POST upload');
 
-  var prefix = '';
-  switch (req.body.type) {
-    case uploader.FileType.NETWORK:
-      prefix = networkPath;
-      break;
-    case uploader.FileType.BINDING:
-      prefix = bindingPath;
-      break;
-    case uploader.FileType.EXPRESSION:
-      prefix = expressionPath;
-      break;
-    case uploader.FileType.BED:
-      prefix = bedPath;
-      break;
-    case uploader.FileType.MAPPING:
-      prefix = mappingPath;
-      break;
-  }
   var body = {
     type: req.body.type,
-    name: req.body.name,
+    dataName: req.body.name,
     description: req.body.description
   };
-  uploader.uploadFile(body, req.file, prefix, bigWigToWigPath, uploadPath);
-  res.header('Access-Control-Allow-Origin', '*');
-  res.json({
-    success: true
+  var data = {};
+  dbData.createConnection(function(db, err) {
+    uploader.uploadFile(body, req.file, dataPath, bigWigToWigPath, db,
+      function(err) {
+        data = err;
+      });
   });
+  res.header('Access-Control-Allow-Origin', '*');
+  if (data.error) {
+    res.status(500).json(data.error);
+  } else {
+    res.json({
+      success: true
```

Comment:
So we are going to use that function? Should I define it outside app.get/app.post?